src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -1207,26 +1381,47 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
               +(1._r8-frac_sno(c)-frac_h2osfc(c))*t_soisno(c,1)**4 &
               +frac_h2osfc(c)*t_h2osfc(c)**4)
 
-         err(p) = (1.0_r8-frac_rad_abs_by_stem)*(sabv(p) + air(p) + bir(p)*tlbef(p)**3 &
+         err(p) = (1.0_r8-frac_rad_abs_by_stem(p))*(sabv(p) + air(p) + bir(p)*tlbef(p)**3 &
               *(tlbef(p) + 4._r8*dt_veg(p)) + cir(p)*lw_grnd) &
-                - lw_leaf + lw_stem - eflx_sh_veg(p) - hvap*qflx_evap_veg(p) &
-                - ((t_veg(p)-tl_ini)*cp_leaf/dtime)
+                - lw_leaf(p) + lw_stem(p) - eflx_sh_veg(p) - hvap*qflx_evap_veg(p) &
+                - ((t_veg(p)-tl_ini(p))*cp_leaf(p)/dtime)
+
+         !  Update stem temperature; adjust outgoing longwave
+         !  does not account for changes in SH or internal LW,
+         !  as that would change result for t_veg above
+         if (use_biomass_heat_storage) then
+            if (stem_biomass(p) > 0._r8) then
```

Comment:
for compatability with FATES, we would also have to have separate code here

Reply:
I don't actually think any of these terms will be different in the FATES case. Most of these terms are from the atmosphere or surface or biomass heating terms. Getting the biomass for stem and leaf are the things that we need from FATES.

Reply:
Because I opened #1247. I'm resolving this conversation.